### PR TITLE
Exercise 2

### PR DESCRIPTION
**Pull request by Kelliep11**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: README.md
Comment (JackMeadDev) [inline README.md:37]: It would be good to also add the keys (but not the values) for these two to the `.env.template` file so that there's another hint there to remember to set them

File: test_my_code.py
Comment (JackMeadDev) [inline test_my_code.py:1]: This looks like a really useful learning file, but not one relevant to our codebase! We'll replace these with more relevant tests in M3 anyway, so I probably suggest leaving it for now and removing it then, but generally we want to keep this codebase tightly focused on the project otherwise it'll quickly get harder to navigate

File: todo_app/data/trello_items.py
Comment (JackMeadDev) [inline todo_app/data/trello_items.py]: I think you've half-converted this to a new system, but not quite finished - the below change should fix up the issue that this currently has where you aren't passing through the idList argument from `app.py`

```suggestion
def add_card(name):
    
    url = "https://api.trello.com/1/cards"

    querystring = {
        "key":os.getenv("TRELLO_API_KEY"),
        "token":os.getenv("TRELLO_API_TOKEN"),
        "name":name,
        "idList":"620e338c101c43271f9ac494"
    }
```

File: todo_app/data/trello_items.py
Comment (JackMeadDev) [inline todo_app/data/trello_items.py]: Although this id (and your list ids below) are not sensitive, it would still be nice to extract them so that your app is configurable for other people's Trello boards - currently we can _only_ run your app against your specific Trello board, rather than e.g. having different boards in different environments. Pulling those out into environment variables would be a nice touch (and similarly with any hard-coded list ids!)

```suggestion
    url = f"https://api.trello.com/1/boards/{os.getenv('TRELLO_BOARD_ID')}/lists"
```

File: test_my_code.py
Comment (hugh-emerson) [inline test_my_code.py:1]: You can delete this file now

File: todo_app/data/trello_items.py
Comment (JackMeadDev) [inline todo_app/data/trello_items.py:65]: Are these print statements useful? They often are when developing/debugging, but they might be worth now removing unless their additional logging continues to still be worthwhile

**Review by JackMeadDev — APPROVED** [review]
Nice work, I needed to tweak one small mistake in your code where it looks like you've switched between two methods of passing round the `idList` parameters without quite lining everything up, but otherwise it was easy to run your app following the instructions and storing/viewing/updating my to-dos in Trello 🙂 
I've left some additional optional suggestions below, but I'm happy for you to merge this when you're ready

File: todo_app/data/trello_items.py
Comment (hugh-emerson) [inline todo_app/data/trello_items.py]: This works but it would be nice to make the board id into an environment variable.

It's not sensitive but it is something we want to vary between environments - e.g. developers running the app against their own boards, a shared test environment and a "real" production environment.
It would also help with automated end-to-end testing.

And then the same applies to the two list IDs used lower down.

File: todo_app/data/trello_items.py
Comment (hugh-emerson) [inline todo_app/data/trello_items.py]: The conventional name for a Python variable would be `id_list`

File: todo_app/data/trello_items.py
Comment (hugh-emerson) [inline todo_app/data/trello_items.py:49]: You may want to remove print statements before merging. Stop them building up and getting noisy.

File: README.md
Comment (hugh-emerson) [inline README.md:37]: You could put these two variables into the .env.template file with placeholder values to make it slightly simpler for the reader to get set up.

And perhaps common sense, but the reader needs a Trello account first.

**Review by hugh-emerson — APPROVED** [review]
Nice work! I've left some minor suggestions but merge when you're ready 👍 

EDIT: or just ignore me and look at Jack's review above. In particular, the id_List function parameter that needs to be removed.

File: README.md
Comment (Kelliep11) [reply README.md:37]: Have added the keys to the .env.template file

File: test_my_code.py
Comment (Kelliep11) [reply test_my_code.py:1]: deleted this file

File: todo_app/data/trello_items.py
Comment (Kelliep11) [reply todo_app/data/trello_items.py:65]: have removed unnecessary prints

File: test_my_code.py
Comment (Kelliep11) [reply test_my_code.py:1]: Removed file

File: todo_app/data/trello_items.py
Comment (Kelliep11) [reply todo_app/data/trello_items.py]: Have made this a variable as per Jacks suggestion

File: todo_app/data/trello_items.py
Comment (Kelliep11) [reply todo_app/data/trello_items.py:49]: Print statements removed

File: README.md
Comment (Kelliep11) [reply README.md:37]: Have added this to .env.template file

**Pull request merged**
